pkg/services/store/entity/db/dbimpl/dbEngine.go

```diff
@@ -1,75 +1,98 @@
 package dbimpl
 
 import (
+	"cmp"
 	"fmt"
 	"strings"
 	"time"
 
-	"github.com/grafana/grafana/pkg/infra/tracing"
-	"github.com/grafana/grafana/pkg/services/sqlstore"
-	"github.com/grafana/grafana/pkg/setting"
-	"github.com/grafana/grafana/pkg/util"
+	"github.com/go-sql-driver/mysql"
 	"xorm.io/xorm"
+
+	"github.com/grafana/grafana/pkg/infra/tracing"
+	"github.com/grafana/grafana/pkg/services/store/entity/db"
 )
 
-func getEngineMySQL(cfgSection *setting.DynamicSection, tracer tracing.Tracer) (*xorm.Engine, error) {
-	dbHost := cfgSection.Key("db_host").MustString("")
-	dbName := cfgSection.Key("db_name").MustString("")
-	dbUser := cfgSection.Key("db_user").MustString("")
-	dbPass := cfgSection.Key("db_pass").MustString("")
+func getEngineMySQL(getter *sectionGetter, _ tracing.Tracer) (*xorm.Engine, error) {
```

Comment:
There are a few minor differences between this implementation and the previous one:
- sql_mode: @@SESSION.sql_mode": "TRADITIONAL,ANSI"
- Loc: time.UTC
- New config flags :  db_server_pub_key / db_tls_config_name
- CheckConnLiveness = true

I suggest we keep this fully backward compatible for now.

Reply:
Reduced the items being set to the bare minimum, moved to TODO comments the rest.